spec/ddtrace/utils_spec.rb

```diff
@@ -0,0 +1,141 @@
+require 'concurrent-ruby'
+
+RSpec.describe Datadog::Utils do
+  describe '.next_id' do
+    subject(:next_id) { described_class.next_id }
+
+    it 'returns a positive integer smaller than 2**62' do
+      is_expected.to be_a(Integer)
+      is_expected.to be_between(1, 2**62 - 1)
+    end
+
+    it 'fits in a CRuby VALUE slot', if: Datadog::Core::Environment::ObjectSpace.estimate_bytesize_supported? do
+      expect(Datadog::Core::Environment::ObjectSpace.estimate_bytesize(next_id)).to eq(0)
+    end
+
+    it 'returns unique numbers on successive calls' do
+      # This is the same assumption of non-collision we have in our production systems.
+
+      ids = Array.new(100) { described_class.next_id }.uniq
+      expect(ids).to have(100).items
+    end
```

Comment:
> I think there's an interesting property being tested here: that we don't mishandle the rng. So my suggestion would be to simplify this test and actually only generate two numbers. If two are different, then the RNG is probably working fine -- no need to generate many more to test.

I can do that.

> This test is inherently flaky, because it's not impossible that the same number would be generated twice (we're playing the birthday problem here).

Technically correct, but we deemed this acceptable across all of APM.